trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,301 @@
+        full_view = self._get_full_table(view_name, schema)
+        query = f'SHOW CREATE VIEW {full_view}'
+        try:
+            res = connection.execute(sql.text(query))
+            return res.scalar()
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+            ):
+                raise exc.NoSuchTableError(full_view) from e
+            raise
+
+    def get_indexes(self, connection: Connection,
+                    table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        if not self.has_table(connection, table_name, schema):
+            raise exc.NoSuchTableError(f'schema={schema}, table={table_name}')
+
+        partitioned_columns = self._get_columns(connection, f'{table_name}$partitions', schema, **kw)
+        partition_index = dict(
+            name='partition',
+            column_names=[col['name'] for col in partitioned_columns],
+            unique=False
+        )
+        return [partition_index, ]
+
+    def get_unique_constraints(self, connection: Connection,
+                               table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        """Trino has no support for unique constraints. Returns an empty list."""
+        return []
+
+    def get_check_constraints(self, connection: Connection,
+                              table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        """Trino has no support for check constraints. Returns an empty list."""
+        return []
+
+    def get_table_comment(self, connection: Connection,
+                          table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
+        properties_table = self._get_full_table(f"{table_name}$properties", schema)
+        query = f'SELECT "comment" FROM {properties_table}'
+        try:
+            res = connection.execute(sql.text(query))
+            return dict(text=res.scalar())
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.NOT_FOUND,
+                    error.COLUMN_NOT_FOUND,
+                    error.TABLE_NOT_FOUND,
+            ):
+                return dict(text=None)
+            raise
+
+    def has_schema(self, connection: Connection, schema: str) -> bool:
+        query = f"SHOW SCHEMAS LIKE '{schema}'"
+        try:
+            res = connection.execute(sql.text(query))
+            return res.first() is not None
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+            ):
+                return False
+            raise
+
+    def has_table(self, connection: Connection,
```

Comment:
The API docs say that the table being searched is an exact table instead of a pattern - so we should be escaping the `table_name` properly here.

Same in `has_schema`.

Refer to the postgresql dialect here: https://github.com/sqlalchemy/sqlalchemy/blob/93c8f1df8bb68820624559506f6713a16a9e2250/lib/sqlalchemy/dialects/postgresql/base.py#L3302-L3355

Reply:
Yeah, but LIKE statement don't use pattern. So it's exactly match.

FYI, I already changed to using `information_schema` instead of `SHOW`.

Reply:
`LIKE 'a_table'` will match `abtable` too.

Also, I still see SHOW TABLES here - did you forget to push some commits?

Reply:
Nevermind, see it's fixed in next commit.

Reply:
All commits are pushed, could you see it?